code/builders/builder-vite/src/vite-config.ts

```diff
@@ -77,6 +77,11 @@ export async function commonConfig(
 
 export async function pluginConfig(options: Options) {
   const frameworkName = await getFrameworkName(options);
+  const { features } = options;
+
+  if (features?.fastBuildEmptyBlocks) {
+    globals['@storybook/blocks'] = '__STORYBOOK_BLOCKS_EMPTY_MODULE__';
```

Comment:
This is a bit of a hack, reusing the `externalGlobalsPlugin` for this purpose.

It would be more "correct" and easier to understand later to create a separate plugin for this that did

```
// before
import { Primary, Stories } from '@storybook/blocks';

//after
const Primary = undefined, Stories = undefined;
```

effectively it's the same, but I think this might be easier to debug down the line. It's also more effort though, which might not be worth it right now.